src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -617,19 +666,41 @@ impl SimulationElement for A320EngineDrivenPumpController {
             panic!("The A320 only supports two engines.");
         }
     }
+
+    fn receive_power(&mut self, supplied_power: &SuppliedPower) {
+        if self.engine_number == 1 {
+            self.is_powered = supplied_power
+                .potential_of(&self.powering_buses[0])
+                .is_powered();
+        } else if self.engine_number == 2 {
+            self.is_powered = supplied_power
+                .potential_of(&self.powering_buses[0])
+                .is_powered()
+                || supplied_power
+                    .potential_of(&self.powering_buses[1])
+                    .is_powered();
+        } else {
+            panic!("The A320 only supports two engines.");
+        }
+    }
 }
 
 struct A320BlueElectricPumpController {
+    is_powered: bool,
+    powering_bus: ElectricalBusType,
     should_pressurise: bool,
     has_pressure_low_fault: bool,
     is_pressure_low: bool,
     weight_on_wheels: bool,
 }
 impl A320BlueElectricPumpController {
     const MIN_ENGINE_OIL_PRESS_THRESHOLD_TO_INHIBIT_FAULT: f64 = 18.;
+    const POWER_BUS: ElectricalBusType = ElectricalBusType::DirectCurrentEssential;
```

Comment:
For consistency I'd still suggest injecting this into the type. What do you think?

In comments below I've written "As above".

Reply:
Did that for the EDP, while still having the buses as constant in the A320Hydraulics type. Just to have those definition at the same place instead of having various buses names spread around in the constructors calls